flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/rest/DeployScriptITCase.java

```diff
@@ -0,0 +1,348 @@
+    @Order(1)
+    public static final SqlGatewayServiceExtension SQL_GATEWAY_SERVICE_EXTENSION =
+            new SqlGatewayServiceExtension(Configuration::new);
+
+    @RegisterExtension
+    @Order(2)
+    private static final SqlGatewayRestEndpointExtension SQL_GATEWAY_REST_ENDPOINT_EXTENSION =
+            new SqlGatewayRestEndpointExtension(SQL_GATEWAY_SERVICE_EXTENSION::getService);
+
+    private static TestingRestClient restClient;
+    private static SessionHandle sessionHandle;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        restClient = TestingRestClient.getTestingRestClient();
+        sessionHandle =
+                new SessionHandle(
+                        UUID.fromString(
+                                restClient
+                                        .sendRequest(
+                                                SQL_GATEWAY_REST_ENDPOINT_EXTENSION
+                                                        .getTargetAddress(),
+                                                SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetPort(),
+                                                OpenSessionHeaders.getInstance(),
+                                                EmptyMessageParameters.getInstance(),
+                                                new OpenSessionRequestBody(
+                                                        "test",
+                                                        Collections.singletonMap("key", "value")))
+                                        .get()
+                                        .getSessionHandle()));
+    }
+
+    @Test
+    void testDeployScriptToYarnCluster() throws Exception {
+        verifyDeployScriptToCluster("yarn-application");
+    }
+
+    @Test
+    void testDeployScriptToKubernetesCluster() throws Exception {
+        verifyDeployScriptToCluster("kubernetes-application");
+    }
+
+    private void verifyDeployScriptToCluster(String target) throws Exception {
+        TestApplicationClusterClientFactory.id = target;
+        String script =
+                "CREATE TEMPORARY TABLE sink(\n"
+                        + "  a INT\n"
+                        + ") WITH (\n"
+                        + "  'connector' = 'blackhole'\n"
+                        + ")\n"
+                        + "INSERT INTO sink VALUES (1);";
+        assertThat(
+                        restClient
+                                .sendRequest(
+                                        SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetAddress(),
+                                        SQL_GATEWAY_REST_ENDPOINT_EXTENSION.getTargetPort(),
+                                        DeployScriptHeaders.getInstance(),
+                                        new SessionMessageParameters(sessionHandle),
+                                        new DeployScriptRequestBody(
+                                                script,
+                                                null,
```

Comment:
It’s recommended to pass a test ExecutionConfig here instead of null to verify whether executionConfig takes effect correctly.